sites/example-project/src/components/viz/DataTable.svelte

```diff
@@ -1,356 +1,602 @@
-                  "unavailable"}
-                evidenceTypeFidelity={column.evidenceColumnType?.typeFidelity ||
-                  "unavailable"}
-              >
-                {column.title}
-              </th>
-            {/each}
-          </tr><tr />
-        </thead>
-        <tbody>
-          {#each dataPage as row, i}
-            <tr
-              class="data"
-              style={rowLines === "false" || i === dataPage.length - 1
-                ? ""
-                : "border-bottom: thin solid rgb(231, 231, 231);"}
-            >
-              {#if rowNumbers === "true"}
-                <td class="index" style="width:2%">
-                  {#if i === 0}
-                    {(index + i + 1).toLocaleString()}
-                  {:else}
-                    {(index + i + 1).toLocaleString()}
-                  {/if}
-                </td>
+              {#if rowNumbers}
+              <th 
+                  class="index" 
+                  style="
+                      width:2%;
+                      background-color: {headerColor};
+                      "></th>
               {/if}
-              {#each Object.values(row) as cell, j}
-                {#if cell == null}
-                  <td
-                    class="null {columnSummary[j].type}"
-                    style="width:{columnWidths}%"
+          {#if $props.columns.length > 0}
+              {#each $props.columns as column, i}
+                  <th
+                      class="{columnSummary.filter(d => d.id === column.id)[0].type}"
+                      style="
+                      text-align: {column.align};
+                      color: {headerFontColor};
+                      background-color: {headerColor};
+                      cursor: {sortable ? 'pointer' : 'auto'};
+                      "
+                      on:click={sortable ? sort(column.id) : ''}
                   >
-                    {"Ø"}
-                  </td>
-                {:else if columnSummary[j].type === "number"}
-                  <td class="number" style="width:{columnWidths}%;">
-                    {formatValue(
-                      cell,
-                      columnSummary[j].format,
-                      columnSummary[j].columnUnitSummary
-                    )}
-                  </td>
-                {:else if columnSummary[j].type === "date"}
-                  <td
-                    class="string"
-                    style="width:{columnWidths}%"
-                    title={formatValue(cell, columnSummary[j].format)}
+                      {column.title ? column.title : formatColumnTitles ? columnSummary.filter(d => d.id === column.id)[0].title : columnSummary.filter(d => d.id === column.id)[0].id}
+                      {#if sortBy.col === column.id}
+                          <span class=icon-container>
+                              {#if sortBy.ascending}
+                                  <span class=sort-icon>
```

Comment:
I found it helpful to abstract this sort icon into a component in a customer project.

This helped with the vertical alignment issue:

In the table:
```svelte
<th class="name"><button on:click={sort("name")}>Customer<SortIcon {sortBy} column="name" /></button></th>
```

And the component


```svelte
<script>
    import TiArrowUnsorted from "svelte-icons/ti/TiArrowUnsorted.svelte";
    import TiArrowSortedUp from "svelte-icons/ti/TiArrowSortedUp.svelte";
    import TiArrowSortedDown from "svelte-icons/ti/TiArrowSortedDown.svelte";

    export let sortBy;
    export let column;
</script>

<div class="icon-container">
    {#if sortBy.col === column}
        {#if sortBy.ascending}
            <div class="sort-icon">
                <!-- <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="s-zV66bAK704iV"><path d="M18.2 13.3l-6.2-6.3-6.2 6.3c-.2.2-.3.5-.3.7s.1.5.3.7c.2.2.4.3.7.3h11c.3 0 .5-.1.7-.3.2-.2.3-.5.3-.7s-.1-.5-.3-.7z"></path></svg> -->
                <TiArrowSortedUp />
            </div>
        {:else}
            <div class="sort-icon">
                <!-- <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="s-zV66bAK704iV"><path d="M5.8 9.7l6.2 6.3 6.2-6.3c.2-.2.3-.5.3-.7s-.1-.5-.3-.7c-.2-.2-.4-.3-.7-.3h-11c-.3 0-.5.1-.7.3-.2.2-.3.4-.3.7s.1.5.3.7z"></path></svg> -->
                <TiArrowSortedDown />
            </div>
        {/if}
    {:else}
        <div class="sort-icon">
            <!-- <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="s-zV66bAK704iV"><path d="M18.2 13.3l-6.2-6.3-6.2 6.3c-.2.2-.3.5-.3.7s.1.5.3.7c.2.2.4.3.7.3h11c.3 0 .5-.1.7-.3.2-.2.3-.5.3-.7s-.1-.5-.3-.7z"></path></svg> -->
            <TiArrowUnsorted />
        </div>
    {/if}
    </div>

<style>
    .sort-icon {
        width: 12px;
        height: 12px;
        align-items: center;
    }

    .icon-container {
        display: inline-flex;
        align-items: center;
        vertical-align: bottom;
        margin-bottom: 0.6em;
        margin-left: 0.2em;
    }
</style>
```